src/pkg/backup/details/testdata/testdata.go

```diff
@@ -33,7 +33,97 @@ func mustAppendPath(p path.Path, newElement string, isItem bool) path.Path {
 	return newP
 }
 
+func locFromRepo(rr path.Path, isItem bool) *path.Builder {
+	loc := &path.Builder{}
+
+	for _, e := range rr.Folders() {
+		loc = loc.Append(strings.TrimSuffix(e, folderSuffix))
+	}
+
+	if rr.Service() == path.OneDriveService || rr.Category() == path.LibrariesCategory {
+		loc = loc.PopFront()
+	}
+
+	// Folders don't have their final element in the location.
+	if !isItem {
+		loc = loc.Dir()
+	}
+
+	return loc
+}
+
+type repoRefAndLocRef struct {
+	RR  path.Path
+	loc *path.Builder
+}
+
+func (p repoRefAndLocRef) mustAppend(newElement string, isItem bool) repoRefAndLocRef {
+	e := newElement + folderSuffix
+
+	if isItem {
+		e = newElement + fileSuffix
+	}
+
+	res := repoRefAndLocRef{
+		RR: mustAppendPath(p.RR, e, isItem),
+	}
+
+	res.loc = locFromRepo(res.RR, isItem)
+
+	return res
+}
+
+func (p repoRefAndLocRef) ItemLocation() string {
+	return strings.TrimSuffix(p.RR.Item(), fileSuffix)
+}
+
+func (p repoRefAndLocRef) FolderLocation() string {
+	lastElem := p.RR.ToBuilder().LastElem()
+
+	if len(p.RR.Item()) > 0 {
+		f := p.RR.Folders()
+		lastElem = f[len(f)-2]
+	}
+
+	return p.loc.Append(strings.TrimSuffix(lastElem, folderSuffix)).String()
+}
+
+func mustPathRep(ref string, isItem bool) repoRefAndLocRef {
+	res := repoRefAndLocRef{}
+	tmp := mustParsePath(ref, isItem)
+
+	// Now append stuff to the RepoRef elements so we have distinct LocationRef
+	// and RepoRef elements to simulate using IDs in the path instead of display
+	// names.
```

Comment:
I'm on board with the future this sets up, but do you have any concerns about it allowing current cases that depend on identical naming to slip past undetected?

Reply:
I think getting tests in selectors and the CLI to use this is probably "good enough" for the moment as those are the things that most heavily exercise different paths. There will likely be more changes in the future for details merging, but we'll get there when we actually change the RepoRef to something else. Doing those changes now would disconnect them from the reason it's being changed